rlberry/utils/check_agent.py

```diff
@@ -0,0 +1,202 @@
+from rlberry.envs import Chain
+from rlberry.envs.benchmarks.ball_exploration import PBall2D
+from rlberry.manager import AgentManager
+import numpy as np
+from rlberry.seeding import set_external_seed
+import tempfile
+import os
+
+SEED = 42
+
+
+def check_agent_manager(Agent, continuous_state=False):
+    """
+    Check that the agent is compatible with :class:`~rlberry.manager.AgentManager`.
+
+    Parameters
+    ----------
+
+    Agent: rlberry agent module
+        Agent class that we want to test.
+
+    continuous_state: bool, default=False
+        whether we test on a discrete state environment or a continuous one.
+    """
+    if continuous_state:
+        env_ctor = PBall2D
+    else:
+        env_ctor = Chain
+    env_kwargs = {}
+    try:
+        agent = AgentManager(
+            Agent, (env_ctor, env_kwargs), fit_budget=5, n_fit=1, seed=SEED
+        )
+        agent.fit()
+    except Exception as exc:
+        raise RuntimeError("Agent not compatible with Agent Manager") from exc
+
+    return agent
+
+
+def check_agent_almost_equal(state, agent1, agent2, n_checks=5):
+    """
+    Check that two agents are almost equal in the sense that they give the same
+    actions n_checks times on a given state.
+
+
+    Parameters
+    ----------
+
+    state: state
+        environment state
+    agent1: rlberry agent
```

Comment:
Agent or agent manager? It seems that you are using a manager below.

Reply:
You are right, agent manager.